community/kernel/src/test/java/org/neo4j/kernel/api/index/IndexAccessorCompatibility.java

```diff
@@ -194,6 +194,20 @@ public void testIndexSeekByPrefixOnNonStrings() throws Exception
         }
     }
 
+    protected List<Long> getAllNodesFromIndexSeekByContains( String term ) throws IOException
+    {
+        try ( IndexReader reader = accessor.newReader() )
+        {
+            List<Long> list = new LinkedList<>();
```

Comment:
Other kernel methods use linked list. It would be good to consider refactoring them all together, but only together with proper micro-benchmarks.